applications/GeoMechanicsApplication/tests/cpp_tests/custom_elements/test_integration_schemes.cpp

```diff
@@ -0,0 +1,130 @@
+{
+    constexpr auto tolerance = 1.0e-6;
+
+    auto contains_lower_bound_of_xi = [tolerance](const auto& rPoint) {
+        return std::abs(rPoint[0] + 1.0) <= tolerance;
+    };
+    KRATOS_EXPECT_TRUE(std::any_of(rIntegrationPoints.begin(), rIntegrationPoints.end(), contains_lower_bound_of_xi))
+
+    auto contains_upper_bound_of_xi = [tolerance](const auto& rPoint) {
+        return std::abs(rPoint[0] - 1.0) <= tolerance;
+    };
+    KRATOS_EXPECT_TRUE(std::any_of(rIntegrationPoints.begin(), rIntegrationPoints.end(), contains_upper_bound_of_xi))
+}
+
+void ExpectLocalCoordinatesAreInRange(const Geo::IntegrationPointVectorType& rIntegrationPoints)
+{
+    constexpr auto tolerance = 1.0e-6;
+
+    auto xi_is_in_range = [tolerance](const auto& rPoint) {
+        return std::abs(rPoint[0]) - 1.0 <= tolerance;
+    };
+    KRATOS_EXPECT_TRUE(std::all_of(rIntegrationPoints.begin(), rIntegrationPoints.end(), xi_is_in_range))
+
+    auto non_xi_coordinates_must_be_near_zero = [tolerance](const auto& rPoint) {
+        return (std::abs(rPoint[1]) <= tolerance) && (std::abs(rPoint[2]) <= tolerance);
+    };
+    KRATOS_EXPECT_TRUE(std::all_of(rIntegrationPoints.begin(), rIntegrationPoints.end(),
+                                   non_xi_coordinates_must_be_near_zero))
+}
+
+} // namespace
+
+namespace Kratos::Testing
+{
+
+KRATOS_TEST_CASE_IN_SUITE(ALobattoIntegrationSchemeIsAnIntegrationScheme, KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    constexpr auto number_of_points = 2;
+    const auto     scheme           = LobattoIntegrationScheme{number_of_points};
+
+    KRATOS_EXPECT_NE(dynamic_cast<const IntegrationScheme*>(&scheme), nullptr);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(NumberOfIntegrationPointsMatchesTheNumberOfPointsGivenAtConstructionTime,
+                          KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    const auto supported_numbers_of_points = std::vector<std::size_t>{2, 3};
+
+    for (auto number : supported_numbers_of_points) {
+        const auto scheme = MakeLobattoIntegrationScheme(number);
+
+        KRATOS_EXPECT_EQ(scheme->GetNumberOfIntegrationPoints(), number);
+        KRATOS_EXPECT_EQ(scheme->GetIntegrationPoints().size(), number);
+    }
+}
+
+KRATOS_TEST_CASE_IN_SUITE(SumOfIntegrationPointWeightsOfAllSupportedLobattoSchemesEqualsTwo,
+                          KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    const auto supported_numbers_of_points = std::vector<std::size_t>{2, 3};
+
+    constexpr auto relative_tolerance = 1.0e-6;
+    for (auto number : supported_numbers_of_points) {
+        const auto scheme = MakeLobattoIntegrationScheme(number);
+
+        KRATOS_EXPECT_RELATIVE_NEAR(SumOfWeights(scheme->GetIntegrationPoints()), 2.0, relative_tolerance)
+    }
+}
+
+KRATOS_TEST_CASE_IN_SUITE(PointsOfAllSupportedLobattoSchemesMustBeInRangeAndIncludeBounds,
+                          KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    const auto supported_numbers_of_points = std::vector<std::size_t>{2, 3};
```

Comment:
This is the 3rd time this vector is defined. Maybe a single definition in a helper will avoid work when we start to allow higher Lobatto schemes.

Reply:
Good suggestion. I have extracted a helper function that returns a vector of supported numbers of points for Lobatto integration.